### PR TITLE
Building out API Endpoints

### PR DESCRIPTION
**Pull request by Jredish11**
Did this break anything?
- [x] No
- [ ]  Yes
Type of change
- [ ] New feature (non-breaking change which adds functionality)
- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Skip all the other stuff and briefly explain the fix.
Checklist:
- [ ]  If this code needs to be tested, all tests are passing
- [x]  The code runs locally
- [x]  There are comments where clarification/ organization is needed
- [ ]  The code is DRY. If it's not, I tried my best
- [x]  I reviewed my code before pushing
Summary:
This PR includes functionality of endpoints for Post and Delete requests of subscriptions.  It also includes seed data. Tested endpoint requests in Postman and got the expected responses.  All tests passing, some sad paths included.  Had issue writing sad paths for Delete request.  Need to come back to sad path, edge case tests for these requests.

### CODE REVIEW

File: app/controllers/api/v1/subscriptions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class Api::V1::SubscriptionsController < ApplicationController
+  def create
+    if subscription_params[:customer_id].present? && subscription_params[:tea_id].present?
```
Comment (Jredish11) [inline app/controllers/api/v1/subscriptions_controller.rb:3]: right now this allows one sad path to pass, I was having issues if I did have one customer or tea id included.  it was working like it should In postman but was giving errors locally due to subscription not being able to be created with ids that don't exist.

File: spec/requests/api/v1/subscriptions_spec.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        expect(sub_data[:data][:attributes][:customer_id]).to be_an(Integer)
+
+        expect(sub_data[:data][:attributes]).to have_key(:tea_id)
+        expect(sub_data[:data][:attributes][:tea_id]).to be_an(Integer)
+      end
+    end
+    
+    describe "sad path" do
+      it 'returns 422 with invalid attributes if customer and/or tea does not exist' do
+        invalid_attrs = {
```
Comment (Jredish11) [inline spec/requests/api/v1/subscriptions_spec.rb:65]: sad path works if I don't include the attributes but if I try to put in an customer id that doesn't exist it will not pass the test locally. when I put a non existant customer id in postman raw body it gives me the expected response though.

File: spec/requests/api/v1/subscriptions_spec.rb [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      it "deletes a subscription" do
+        subscription = create(:subscription)
+        
+        delete "/api/v1/subscriptions/#{subscription.id}"
+
+        expect(response).to be_successful
+        data = JSON.parse(response.body, symbolize_names: true)
+        expect(data).to be_a(Hash)
+        expect(data).to have_key(:message)
+        expect(data[:message]).to eq("Subscription successfully deleted")
```
Comment (Jredish11) [inline spec/requests/api/v1/subscriptions_spec.rb:88]: due to the same issue I had with the other sad path, I wasn't able to figure out how to write a sad path locally.  but it did work in postman. I would delete a subscriptions and then try to delete the same one but got back the 404 error.
